### PR TITLE
feat: Support multi platforms build

### PR DESCRIPTION
**Pull request by maiquanghiep**

### CODE REVIEW

**Review by liam-icheng-lai — APPROVED** [review]

File: .github/workflows/reusable_docker_pipeline.yml [hunk 1]
```diff
@@ -25,33 +25,74 @@ on:
...
+    steps:
+      - name: Prepare
+        run: |
+          platform=${{ matrix.platform }}
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
+  
       - name: Checkout repository
         uses: actions/checkout@v4
 
+      - name: Set up QEMU
```
Comment (filippos47) [inline .github/workflows/reusable_docker_pipeline.yml:75]: Why do we need QEMU?

File: .github/workflows/reusable_docker_pipeline.yml [hunk 2]
```diff
@@ -67,103 +108,139 @@ jobs:
...
+      - name: Build and push to Docker Hub
+        id: build-dockerhub
+        uses: docker/build-push-action@v6
+        with:
+          platforms: ${{ matrix.platform }}
+          context: ${{ inputs.dockerContext }}
+          file: ${{ inputs.dockerfile }}
+          outputs: type=image,name=${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }},push-by-digest=true,push=${{ inputs.publish }}
+
+      - name: Build and push to ECR
```
Comment (filippos47) [inline .github/workflows/reusable_docker_pipeline.yml:120]: Same here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # CHANGELOG
 
+## 0.5.0
+
+- resuable_docker_pipeline: Add `platforms` input to support Multi-platform image
+- resuable_docker_pipeline: Refractor docker_build job
```
Comment (filippos47) [inline CHANGELOG.md:6]:
```suggestion
- resuable_docker_pipeline: Refactor docker_build job
```

Comment (filippos47) [reply CHANGELOG.md:6]: Also typo on reusable :smile:

File: .github/workflows/reusable_docker_pipeline.yml [hunk 4]
```diff
@@ -25,33 +25,74 @@ on:
...
+        runner:
+          - ubuntu-24.04
+          - ubuntu-24.04-arm64
+        exclude:
+        - platform: linux/amd64
+          runner: ubuntu-24.04-arm64
+        - platform: linux/arm64
+          runner: ubuntu-24.04
+    steps:
+      - name: Prepare
```
Comment (filippos47) [inline .github/workflows/reusable_docker_pipeline.yml:67]: Could this be unified with the `prepare-metadata` job?

File: .github/workflows/reusable_docker_pipeline.yml [hunk 5]
```diff
@@ -67,103 +108,139 @@ jobs:
...
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      
+      - name: Create manifest list and push
+        working-directory: /tmp/digests
+        run: |
+          tags=$(jq -cr '.tags | map("-t " + .) | join(" ")'  <<< "$DOCKER_METADATA_OUTPUT_JSON")
+          digests=$(printf "${{ vars.DOCKERHUB_REGISTRY_ID }}/${{ needs.prepare-metadata.outputs.image-name }}@sha256:%s " $(ls /tmp/digests))
+          docker buildx imagetools create $tags $digests
+      
+      - name: Inspect image
```
Comment (filippos47) [inline .github/workflows/reusable_docker_pipeline.yml:200]: Is this step used to verify that push operation succeeded?

**Review by filippos47 — APPROVED** [review]
Great and important work for the future! :fire:

File: .github/workflows/reusable_docker_pipeline.yml (hunk 4 above)
Comment (maiquanghiep) [reply .github/workflows/reusable_docker_pipeline.yml:67]: This step gets the `platform=${{ matrix.platform }}` which is only available in the `docker_build` job

File: .github/workflows/reusable_docker_pipeline.yml (hunk 1 above)
Comment (maiquanghiep) [reply .github/workflows/reusable_docker_pipeline.yml:75]: QEMU is recommended when building Docker images for different architectures.
Indeed as we now build the image natively so can be removed

File: .github/workflows/reusable_docker_pipeline.yml [hunk 6]
```diff
@@ -67,103 +108,139 @@ jobs:
           echo "EOF" >> $GITHUB_OUTPUT
         shell: bash
 
-      - name: Build Docker image
+      - name: Build and push to Docker Hub
```
[the remark this answers is not in the document]
Comment (maiquanghiep) [reply .github/workflows/reusable_docker_pipeline.yml:111]: We do push to the registry without the tag if `push=${{ inputs.publish }}`. The later job which merges the two digest to the same tag

File: .github/workflows/reusable_docker_pipeline.yml (hunk 5 above)
Comment (maiquanghiep) [reply .github/workflows/reusable_docker_pipeline.yml:197]: Yes, this step only shows the image's information

File: .github/workflows/reusable_docker_pipeline.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (filippos47) [reply .github/workflows/reusable_docker_pipeline.yml:108]: Oh I understand now!

**Review by filippos47 — APPROVED** [review]

**Pull request merged**